lexicon/modification.py

```diff
@@ -185,3 +186,56 @@ def run(self):
                 merged_lex.lemmata.extend(lex.lemmata)
 
         write_xml(self.out_bliss_lexicon.get_path(), merged_lex.to_xml())
+
+
+class AddEowPhonemesToLexiconJob(Job):
+    def __init__(self, bliss_lexicon: tk.Path):
+        """
+        Extends phoneme set of a lexicon by additional end-of-word (eow) versions
+        of all regular phonemes. Modifies lemmata to use the new eow-version
+        of the final phoneme in each pronunciation.
+
+        :param tk.Path bliss_lexicon: Base lexicon to be modified.
+        """
+        self.bliss_lexicon = bliss_lexicon
+
+        self.out_lexicon = self.output_path("lexicon.xml")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        in_lex = lexicon.Lexicon()
+        in_lex.load(self.bliss_lexicon.get_path())
+
+        out_lex = lexicon.Lexicon()
+
+        # Identify all 'special' phonemes in the given lexicon. A phoneme is
+        # deemed 'special' if it appears in the pronunciation of a special lemma.
+        # This should identify e.g. 'silence', 'blank', 'unknown' etc.
```

Comment:
how about `[noise]` and similar?

Reply:
Yes, fixed now by adding a `nonword_phones` parameter to the job where such phonemes can be specified.